addons/mail/models/mail_thread.py

```diff
@@ -3167,15 +3167,17 @@ def _notify_thread_by_inbox(self, message, recipients_data, msg_vals=False, **kw
                 'res_partner_id': pid,
             } for pid in inbox_pids]
             self.env['mail.notification'].sudo().create(notif_create_values)
+            # sudo: res.partner to format personalized message
+            partners = self.env['res.partner'].sudo().browse(inbox_pids)
+            for partner in partners:
+                users = partner.user_ids
+                internal_users = users - users.filtered('share')
+                main_user = (internal_users or users or self.env['res.users'])[:1]
 
-            MailMessage = self.env['mail.message']
-            messages_format_prepared = MailMessage._message_format_personalized_prepare(
-                message._message_format(msg_vals=msg_vals, for_current_user=True), partner_ids=inbox_pids)
-            for partner_id in inbox_pids:
                 bus_notifications.append(
-                    (self.env['res.partner'].browse(partner_id),
+                    (partner,
                      'mail.message/inbox',
-                     MailMessage._message_format_personalize(partner_id, messages_format_prepared)[0])
+                     message.with_user(main_user)._message_format_personalize(partner.id, msg_vals=msg_vals))
```

Comment:
Why this change ? Why do we need a "with_user", as anyway what matter in personalize if the recipient partner ?

Reply:
bug reported: when handling notifications in Odoo and tagging the user on a SO/ task / whatever, the messaging counter is increased by 1 and the notification appears in the inbox, but not in the systray.

reason: previously `            messages_format_prepared = MailMessage._message_format_personalized_prepare(
                message._message_format(msg_vals=msg_vals, for_current_user=True), partner_ids=inbox_pids)
 `in `_message_format` `for_current_user=True` but that is the user that is doing the action, not the user that should be notified.